.github/workflows/set-milestone.yml

```diff
@@ -0,0 +1,36 @@
+#############################################################################
+# GitHub action to set latest milestone on issue of merged PR.
+#
+#############################################################################
+name: 'Set milestone on issue of merged PR'
+
+on:
+  pull_request:
+    types:
+      - closed
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  set-milestone:
+    name: 'Set milestone on issue of merged PR #${{ github.event.pull_request.number }}'
+    if: ${{ github.event.pull_request.merged == true
+      && (startsWith(github.event.pull_request.title, 'Issue')
+      || startsWith(github.event.pull_request.title, 'Pull')) }}
```

Comment:
title is not good source of truth, can we  take it from latest commit of branch ?

Reply:
Why is it not a good source?

Reply:
I will see what is possible.

Reply:
After a bit of digging, we can use `github.event.head_commit.message`. I will apply it.

Reply:
Yes, not validated, usually never updated after creation, as issue number in it is not required.
Issue is commit is source of truth. And it is source for validation of PR description and ...

There might be cases then there is few commits in PR, it is ok to keep marking of only one, most top. It will cover 95% of cases, and 5% will be closed without milestone, not a big deal for now.

Reply:
Uh, turns out `github.event.head_commit.message` is only for direct commits to `master` and does not apply to merged PRs.

Reply:
We still can get latest commit message in script down below, but job will run on every merged PR, even `minor|dependency` and others. We can make it exit with code `0` if that is the case.

Reply:
Done.

PR: https://github.com/stoyanK7/checkstyle/pull/52
Job: https://github.com/stoyanK7/checkstyle/actions/runs/4295695715/jobs/7486461487

Reply:
> latest commit of branch 

If this is going to put milestones only to the issue/pr in the latest commit, then this is not going to work for PRs with more than one commit.

Reply:
We can test this and see how it works, and improve later on if required. Even now we missing a lot milestones, so it is not worse than what we have now.

Reply:
Done. We are now checking all commits for an issue number.